source/read/specify-a-query.txt

```diff
@@ -0,0 +1,265 @@
+      :dedent:
+   
+   .. output::
+      :visible: false
+
+      Fruit(id=2, name=bananas, quantity=7, rating=4, color=yellow, type=[cavendish])
+      Fruit(id=3, name=oranges, quantity=6, rating=2, color=null, type=[naval, mandarin])
+      Fruit(id=4, name=pineapples, quantity=3, rating=5, color=yellow, type=null)
+
+Array Operators
+---------------
+
+Array operators match documents based on the value or quantity of elements in an
+array field. The following is a list of available array operators:
+
+- ``$all``, which returns documents with arrays that contain all elements in the query
+- ``$elemMatch``, which returns documents if an element in their array field matches all conditions in the query
+- ``$size``, which returns all documents with arrays of a specified size
+
+To learn more about array operators, see the :manual:`Array Query Operators
+</reference/operator/query-array/>` guide in the MongoDB Server manual.
+
+The following example specifies an array operator in a query filter as a
+parameter to the ``find()`` method. The code returns all documents with a
+``type`` array field containing exactly ``2`` elements:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/read/specify-a-query.kt
+      :start-after: start-find-array
+      :end-before: end-find-array
+      :language: kotlin
+      :dedent:
+   
+   .. output::
+      :visible: false
+
+      Fruit(id=1, name=apples, quantity=5, rating=3, color=red, type=[fuji, honeycrisp])
+      Fruit(id=3, name=oranges, quantity=6, rating=2, color=null, type=[naval, mandarin])
+
+Element Operators
+-----------------
+
+Element operators query data based on the presence or type of a field.
+
+To learn more about element operators, see the :manual:`Element Query Operators
+</reference/operator/query-element/>` guide in the MongoDB Server manual.
+
+The following example specifies an element operator in a query filter as a
+parameter to the ``find()`` method. The code returns all documents that have a
+``color`` field:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/read/specify-a-query.kt
+      :start-after: start-find-element
+      :end-before: end-find-element
+      :language: kotlin
+      :dedent:
+   
+   .. output::
+      :visible: false
+
+      Fruit(id=1, name=apples, quantity=5, rating=3, color=red, type=[fuji, honeycrisp])
+      Fruit(id=2, name=bananas, quantity=7, rating=4, color=yellow, type=[cavendish])
+      Fruit(id=4, name=pineapples, quantity=3, rating=5, color=yellow, type=null)
+
+Evaluation Operators
+--------------------
+
+Evaluation operators return data based on evaluations of either individual
+fields or the entire collection's documents.
+
+The following is a list of common evaluation operators:
+
+- ``$text``, which performs a text search on the documents
+- ``$regex``, which returns documents that match a specified regular expression
+- ``$mod``, which performs a modulo operation on the value of a field and
```

Comment:
S: I think it could be helpful to link to modulo here for those who don't know what it is
```suggestion
- ``$mod``, which performs a :wikipedia:`modulo </Modulo>` operation on the value of a field and
```